src/constraint/field.ts

```diff
@@ -96,9 +96,10 @@ export const FIELD_CONSTRAINTS: EncodingConstraintModel<FieldQuery>[] = [
     properties: [Property.AGGREGATE, Property.AUTOCOUNT, Property.TIMEUNIT, Property.BIN],
     allowWildcardForProperties: true,
     strict: true,
-    satisfy: (fieldQ: FieldQuery, _: Schema, __: PropIndex<Wildcard<any>>, ___: QueryConfig) => {
+    // TODO(akshatsh): Allow field query with optional autocount?
+    satisfy: (fieldQ: FieldQuery & AutoCountQuery, _: Schema, __: PropIndex<Wildcard<any>>, ___: QueryConfig) => {
```

Comment:
`FieldQuery | AutoCountQuery`, not `FieldQuery & AutoCountQuery`.

Same for a few other places.